### ISSUE TITLE
Consistent object destructuring spacing

### ISSUE THREAD

**Issue by dcousens** [eslint/eslint#2955]
Is it possible a rule could be added for the following cases:

``` javascript
var { x} = ...  // ERROR: Inconsistent spacing
var {x } = ...  // ERROR: Inconsistent spacing
var { x } = ... // OK (1)
var {x} = ... // OK (2)
```

Extra options (though less important for my case) could also be to allow strict enforcement over option 1 or option 2 **ONLY**.

If such a rule already exists,  my apologies.

**Comment by nzakas** [comment]
I believe object-curly-spacing should cover this case: http://eslint.org/docs/rules/object-curly-spacing

**Comment by feross** [comment]
@nzakas object-curly-spacing forces to you pick one style `var {x} = ...` or `var { x } = ...`.

I believe @dcousens is asking for an option to allow *either* style, while preventing the use of inconsistent spacing like `var { x} = ...` or `var {x } = ...`.

**Comment by xjamundx** [comment]
So the proposal is a new option on http://eslint.org/docs/rules/object-curly-spacing#options to be `"either"`?

**Comment by dcousens** [comment]
@xjamundx yes,  this would allow us to enforce consistency,  in an otherwise ambiguous environment where currently anything goes.

**Comment by nzakas** [comment]
Out of curiosity, why not just pick one style?

**Comment by dcousens** [comment]
@nzakas semver.
This would classify as a patch for existing *wrong* behaviour,  but previously we hadn't set the rule as to whether there should be a space or not.

We intend to enforce one eventually, but right now,  it would break too much code (as standard is used for tests).

**Comment by mysticatea** [comment]
The rule is off by default?

**Comment by dcousens** [comment]
@mysticatea I was referring to [standard](https://github.com/feross/standard)'s versioning, not es lint.
This would be a patch for `standard`,  if we enforced one or the other,  it would be a breaking change.

**Comment by nzakas** [comment]
@dcousens it sounds like you're asking us to implement a change that you only intend to use temporarily so as not to require a major version bump for standard right now. I don't think that's a good reason for us to make such a change.

**Comment by dcousens** [comment]
@nzakas I don't think that is true,  as the end result is that we may not enforce 1 style or the other.
For now (probably in the order of months),  we want to enforce this consistently throughout `standard`,  this is the only way we can do that.
I know this change would be beneficial for other projects I work on where we want consistency, but also do not enforce 1 style or the other.

**Comment by lo1tuma** [comment]
ESLint could only enforce consistency within a file. If you don't pick one style you might end with inconsistent styles across all project files.

**Comment by mysticatea** [comment]
We can choose turn the rule on/off each project.

**Comment by dcousens** [comment]
@lo1tuma I'd rather see `{ x }` and `{x}`, consistently,  then anything goes inconsistently.
I didn't mean only 1 style consistently,  I meant both.

>  The end result [may be] that we may don't enforce 1 style XOR** the other.

**Comment by nzakas** [comment]
@dcousens we generally don't have rules that allow two styles at the same time, as this typically is too confusing for users. 

Unless someone on the ESLint team wants to champion this change, I'm leaning towards encouraging you to create a custom rule.

**Comment by dcousens** [comment]
Thanks @nzakas.

@feross shall I close?

**Comment by feross** [comment]
Sure, I think that's totally fair.

**Comment by xjamundx** [comment]
@dcousens, @feross if you want help making a plugin to add this functionality, let me know. it shouldn't be too hard!

**Comment by feross** [comment]
@xjamundx yes, that would be a great help! want to move the discussion over to https://github.com/feross/standard/issues/182 ?
